### PR TITLE
set more def bucket count to get better perfs on huge modules

### PR DESCRIPTION
**Pull request by BBasile**

### CODE REVIEW

**Comment by ZombineDev** [comment]
Perhaps we should change `StringCache.defaultBucketCount`?

**Comment by ZombineDev** [comment]
It would be nice to also add a shorter version of the primes test case and assert that it takes less than a second.

**Comment by BBasile** [comment]
> Perhaps we should change StringCache.defaultBucketCount ?

I have the whole code.dlang.org data locally. I thing i'll compute the average number of tokens per module (only those with string cached, like `tok!"identifier"`)  (note: there are around ~22000 modules). Then the right number will be its `nextPow2()`.

File: src/main.d [hunk 1]
```diff
@@ -169,7 +169,7 @@ else
 
 	immutable usingStdin = args.length == 1;
 
-	StringCache cache = StringCache(StringCache.defaultBucketCount);
+	StringCache cache = StringCache(0x10000);
```
Comment (wilzbach) [inline src/main.d]: Whatever this number will be, imho it should definitely be accompanied by an explanatory comment ;-)

**Comment by BBasile** [comment]
With the time passing i'm less concerned by this issue. There's still a trace of the problem in libdparse issues.

**Pull request closed without merging**